### DISCUSSION TITLE
copy rsa key into remote machine

### DISCUSSION THREAD

**Discussion by reddyarun474** [ktbyers/netmiko#2446] [General]
Hi Team, 

Is there a way to copy id_rsa.pub to remote server.
I saw there is a command to do it and below is the command to do it using ssh
ssh-copy-id -i /root/.ssh/id_rsa.pub root@x.x.x.x

Similarly any API in netmiko to perform it.

**Comment by ktbyers** [comment]
You should be able to execute Linux commands using the Netmiko `linux` device_type so yes you should be able to replicate this.

Note, for Linux automation such as this there are much better tools available than Netmiko (i.e. there are server automation tools that are better suited to these types of operations).

**Reply by reddyarun474** [reply]
Thanks for the response. I will have a look.
